### DISCUSSION TITLE
Let the people have their copilot access, I'm not paying 10$ a month for this.

### DISCUSSION THREAD

**Discussion by TheRedXD** [community/community#19126] [Copilot Conversations] [answered]
I feel like people who have used the copilot technical preview should be allowed to have permanent copilot access, because they have helped making copilot better. This happened exactly as I predicted - we're all surprised for how "cool" this little extension is, and then after a while this gets locked behind a paywall. If you're so supportive of open-source, then maybe, just maybe, you should actually let people use this app for free, who helped make the darn thing. The more users, the more data it can get and therefore it gets better. There was also nothing mentioned about it becoming monetized beforehand. 

This app was actually helpful for me making a lot of really good code, and having to do that from scratch makes me less efficient. If anyone has an excellent alternative, **let me know**. All in all, I'm disappointed in this decision. 10$ might not sound like a lot and you might be screaming at me right now, "omg its just 10 bucks", the economy that we have here, where I live, doesn't exactly *help* me buy some random copilot garbage on the internet, because there is a fairly massive lack of jobs over here.

Not trying again.

Edit: should this even be legal? Copilot is taking github repo code and using it for monetization, but these pieces of code have their own licenses, something like AGPLv3, which could get problematic for things like github who want to make a quick buck from YOUR work by using it for some random AI.

**Comment by CR7273868** [comment]


**Reply by TheRedXD** [reply]
Copilot was actually useful to me; your answer was helpful to no-one, if you don't understand then you might as well not comment at all.

**Reply by CR7273868** [reply]
Aw, good for you. I’ve been using it for a good 8 months now. So, don’t complain, stop using it. I stay by what I said.

**Comment by vedantmgoyal9** [comment, marked as answer]
You can use Tabnine (alternative)

BTW, if you're a student, you can sign up for a GitHub student developer pack... it allows you to use Copilot for free as long as you're a student...

**Reply by TheRedXD** [reply]
Thanks, though I did decide to get github's student developer pack. Tabnine is super helpful tho ngl
